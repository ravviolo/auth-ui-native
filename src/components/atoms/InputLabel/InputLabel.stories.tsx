import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { InputLabel } from './InputLabel';

storiesOf('InputLabel', module)
  .addDecorator((getStory) => <StoryScreen center>{getStory()}</StoryScreen>)
  .add('Default', () => (
    <InputLabel label={text('label', 'Label')} testID={text('testID', 'input-label-test-id')} />
  ));
