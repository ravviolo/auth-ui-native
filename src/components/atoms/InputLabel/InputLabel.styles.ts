import { StyleSheet } from 'react-native';
import { colors } from 'theme';

export const styles = StyleSheet.create({
  label: {
    color: colors.text,
    marginBottom: 3,
  },
});
