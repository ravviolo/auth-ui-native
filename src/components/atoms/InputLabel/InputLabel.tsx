import { StyleSheet, Text } from 'react-native';
import { colors } from 'theme';

export interface Props {
  testID: string;
  label: string;
}

export const InputLabel = ({ label, testID }: Props) => {
  return (
    <Text style={styles.label} testID={testID}>
      {label}
    </Text>
  );
};

const styles = StyleSheet.create({
  label: {
    color: colors.text,
    marginBottom: 3,
  },
});
