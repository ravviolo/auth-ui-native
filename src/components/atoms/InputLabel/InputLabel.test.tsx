import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { InputLabel } from './InputLabel';

import type { Props } from './InputLabel';

describe('InputLabel', () => {
  const props: Props = {
    label: 'Label',
    testID: 'label-test-id',
  };

  it('should display label text received via props', () => {
    render(<InputLabel {...props} />);

    const label = screen.getByTestId('label-test-id');

    expect(label).toHaveTextContent('Label');
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<InputLabel {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
