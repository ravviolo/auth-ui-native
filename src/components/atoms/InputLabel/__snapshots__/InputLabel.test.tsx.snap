// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputLabel should match snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#3D3D3D",
      "marginBottom": 3,
    }
  }
  testID="label-test-id"
>
  Label
</Text>
`;
